extensions/typescript/src/features/referenceProvider.ts

```diff
@@ -14,7 +14,7 @@ export default class TypeScriptReferenceSupport implements ReferenceProvider {
 	public constructor(
 		private client: ITypescriptServiceClient) { }
 
-	public provideReferences(document: TextDocument, position: Position, options: { includeDeclaration: boolean }, token: CancellationToken): Promise<Location[]> {
+	public provideReferences(document: TextDocument, position: Position, options: { includeDeclaration: boolean }, _: (value: Location) => void, token: CancellationToken): Promise<Location[]> {
```

Comment:
Yeah, I agree about not breaking backwards compatibility.

My initial solution involved upgrading the return type to support a `ProgressThenable` (which I don't think would have been backwards incompatible), but initial feedback from the issue I opened suggested that dealing with progress within the context of a promise is non-standard, so callbacks are preferred. I forgot to circle back on this.

The only solution I can think of is to introduce a new interface called `StreamingReferenceProvider` or `ProgressReferenceProvider` with a separate/dedicated register function (thoughts on name?).

Let me know if you have any other ideas.